pkg/metadata/service_metadata.go

```diff
@@ -0,0 +1,118 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package metadata
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/ghodss/yaml"
+)
+
+// ServiceMetadata consists of information about the service and relative links as well
+// as a list of supported/deprecated versions
+type ServiceMetadata struct {
+	Service ServiceDetails `json:"service"`
+	// CRDs to ignore. ACK generator would skip these resources.
+	Versions []ServiceVersion `json:"versions"`
```

Comment:
Let's rename this variable `APIVersions` ? just to avoid confusion with controller image versions (`v0.1.2`... etc)